snapcraft/plugins/v1/flutter.py

```diff
@@ -97,11 +97,12 @@ def pull(self) -> None:
                 check=True,
                 cwd=work_path,
             )
-        subprocess.run(["flutter", "pub", "get"], check=True, cwd=work_path)
 
     def build(self) -> None:
         super().build()
 
+        subprocess.run(["flutter", "pub", "get"], check=True, cwd=self.builddir)
```

Comment:
you can use self.run() which will default the to the context of builddir and share the typical error handling

Reply:
I told ken not to use self.run here as exports in LD_LIBRARY_PATH and such could get in the way